pytorch_toolkit/object_detection/tests/common/test_case.py

```diff
@@ -260,3 +299,109 @@ def test_alt_ssd_export_on_cpu(self):
+            return template_updates_description
+
+        @staticmethod
+        def generate_template_folder_name(src_template_folder, template_updates_description):
+            assert not src_template_folder.endswith('/')
+            template_folder = src_template_folder + '__' +  template_updates_description
+            return template_folder
+
+        @staticmethod
+        def copy_template_folder(src_template_folder, template_folder):
+            logging.info(f'Copying {src_template_folder} to {template_folder}')
+            run_through_shell(f'cp -a "{src_template_folder}" "{template_folder}"')
+            assert os.path.isdir(template_folder), f'Cannot create {template_folder}'
+
+        @staticmethod
+        def apply_update_dict_params_to_template_file(template_file, update_params):
+            template_data = mmcv.Config.fromfile(template_file)
+            template_data.dump(template_file + '.backup.yaml')
+
+            template_data.merge_from_dict(update_params)
+            template_data.dump(template_file)
+
+        def skip_if_cpu_is_not_supported(self):
+            with open(self.template_file) as read_file:
+                training_targets = [x.lower() for x in yaml.load(read_file, yaml.SafeLoader)['training_target']]
+            if 'cpu' not in training_targets:
+                self.skipTest('CPU is not supported.')
+
+        @staticmethod
+        def skip_non_instantiated_template_if_its_allowed(template_folder):
+            # Note that this is for debug purposes only
+            # TODO: refactor, make common function skip_non_instantiated_template_if_its_allowed
+            should_skip_absent_templates = os.environ.get('SHOULD_SKIP_ABSENT_TEMPLATES')
+            if not os.path.isdir(template_folder):
+                if should_skip_absent_templates:
+                    raise unittest.SkipTest(f'The template folder for {problem_name}/{model_name} is not instantiated -- SKIPPING IT')
+                else:
+                    raise unittest.TestCase.failureException(f'The template folder for {problem_name}/{model_name} is not instantiated')
+
+        @unittest.skipUnless(torch.cuda.is_available(), 'No GPU found')
+        def test_nncf_on_gpu(self):
+            log_file = os.path.join(self.template_folder, 'test_nncf.log')
+            run_through_shell(
+                f'cd {self.template_folder};'
+                f'python train.py'
+                f' --train-ann-files {self.ann_file}'
+                f' --train-data-roots {self.img_root}'
+                f' --val-ann-files {self.ann_file}'
+                f' --val-data-roots {self.img_root}'
+                f' --load-weights snapshot.pth'
+                f' --save-checkpoints-to {self.template_folder}'
+                f' --gpu-num 1'
+                f' --batch-size 1'
+                f' |& tee {log_file}')
+            # TODO: think about using |& for other tests
```

Comment:
?

Reply:
Removed.
Now ` | tee ` is used everywhere instead of ` |& tee ` for code unification.